libshortfin/src/shortfin/local/scope.cc

```diff
@@ -82,6 +88,40 @@ std::vector<std::string_view> Scope::device_names() const {
   return names;
 }
 
+Program Scope::LoadUnboundProgram(std::span<const ProgramModule> modules,
+                                  Program::Options options) {
+  std::vector<iree_vm_module_t *> all_modules;
+  std::vector<iree_hal_device_t *> raw_devices;
+
+  // By default, bind all devices in the scope in order to the program.
+  for (Device *d : devices_) {
+    raw_devices.push_back(d->hal_device());
+  }
+
+  // Add a HAL module.
+  iree::vm_module_ptr hal_module;
+  SHORTFIN_THROW_IF_ERROR(iree_hal_module_create(
+      system().vm_instance(), raw_devices.size(), raw_devices.data(),
+      IREE_HAL_MODULE_FLAG_NONE, system().host_allocator(),
+      hal_module.for_output()));
+  all_modules.push_back(hal_module);
+
+  // Add explicit modules.
+  for (auto &pm : modules) {
+    all_modules.push_back(pm.vm_module());
+  }
+
+  // Create the context.
+  iree::vm_context_ptr context;
+  iree_vm_context_flags_t flags = IREE_VM_CONTEXT_FLAG_CONCURRENT;
```

Comment:
heads up that today this mostly just changes the way tracy reports things such that each context gets a fiber track - not sure if you've got tracy running yet but if you see it looking different than normal this is the reason

Reply:
ack